links/links.go

```diff
@@ -0,0 +1,214 @@
+)
+
+// LinkStats is returned from the GetLinkStats, contains an array of Event Stats
+type LinkStats struct {
+	EventStats []EventStats `json:"linkEventStats"`
+}
+
+// EventStats will contain the counts for the aggregations for the requested period
+type EventStats struct {
+	Platform Platform  `json:"platform"`
+	ET       EventType `json:"event"`
+	Count    int32     `json:"count,string"`
+}
+
+// Platform constant "enum" for the event
+type Platform int
+
+// There are 3 possible values for the platforms "enum" in the platform
+const (
+	_ Platform = iota
+	Desktop
+	IOS
+	Android
+)
+
+var platformByID = map[Platform]string{
+	Desktop: "DESKTOP",
+	IOS:     "IOS",
+	Android: "ANDROID",
+}
+
+var platformByName = map[string]Platform{
+	"DESKTOP": Desktop,
+	"IOS":     IOS,
+	"ANDROID": Android,
+}
+
+func (p Platform) String() string {
+	return platformByID[p]
+}
+
+// MarshalJSON makes the "enum" usable in marshalling json
+func (p *Platform) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + platformByID[*p] + `"`), nil
+}
+
+// UnmarshalJSON is used to read the json files into the "enum"
+func (p *Platform) UnmarshalJSON(b []byte) error {
+	var s string
+	err := json.Unmarshal(b, &s)
+	if err != nil {
+		return err
+	}
+	var ok bool
+	if *p, ok = platformByName[s]; !ok {
+		return fmt.Errorf("unknown platform %q", s)
+	}
+	return nil
+}
+
+// EventType constant for the event stats
+type EventType int
+
+// There are 5 possible values for the event type "enum" in the event_stats
+const (
+	_ EventType = iota
+	Click
+	Redirect
+	AppInstall
+	AppFirstOpen
+	AppReOpen
+)
+
+var eventTypesByID = map[EventType]string{
+	Click:        "CLICK",
+	Redirect:     "REDIRECT",
+	AppInstall:   "APP_INSTALL",
+	AppFirstOpen: "APP_FIRST_OPEN",
+	AppReOpen:    "APP_RE_OPEN",
+}
+
+var eventTypesByName = map[string]EventType{
+	"CLICK":          Click,
+	"REDIRECT":       Redirect,
+	"APP_INSTALL":    AppInstall,
+	"APP_FIRST_OPEN": AppFirstOpen,
+	"APP_RE_OPEN":    AppReOpen,
+}
+
+func (e EventType) String() string {
+	return eventTypesByID[e]
+}
+
+// MarshalJSON makes the "enum" usable in marshalling json
+func (e *EventType) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + eventTypesByID[*e] + `"`), nil
+}
+
+// UnmarshalJSON is used to read the json files into the "enum"
+func (e *EventType) UnmarshalJSON(b []byte) error {
+	var s string
+	err := json.Unmarshal(b, &s)
+	if err != nil {
+		return err
+	}
+	var ok bool
+	if *e, ok = eventTypesByName[s]; !ok {
+		return fmt.Errorf("unknown event type %q", s)
+	}
+	return nil
+}
+
+// StatOptions are used in the request for GetLinkStats, it is used to request data
+// going back DurationDays days.
+type StatOptions struct {
+	DurationDays int
+}
+
+// Client is the interface for the Firebase dynamics links service.
+//
+// Client is the entry point to the dynamic links functions
+type Client struct {
+	hc                *internal.HTTPClient
+	linksEndPoint     string
```

Comment:
I'm a little bit confused about why these are members, when it seems their only possible values are the constants above...

Reply:
This one is for testing.
I can move the other one out.